sdk/core/azure-core/src/main/java/com/azure/core/util/Header.java

```diff
@@ -25,7 +35,21 @@
     public Header(String name, String value) {
         Objects.requireNonNull(name, "'name' cannot be null.");
         this.name = name;
-        this.value = value;
+        this.values = new LinkedList<>();
```

Comment:
Any reason a `LinkedList` was chosen over an `ArrayList`? Are we aiming to reduce the number of possible resizes that an `ArrayList` would perform internally? Do we want less memory thrashing or potentially better memory locality here?